// discord.js-commando/src/commands/util/ping.js
const { oneLine } = require('common-tags');
const Command = require('../base');

module.exports = class PingCommand extends Command {
	constructor(client) {
		super(client, {
			name: 'ping',
			group: 'util',
			memberName: 'ping',
			description: 'Checks the bot\'s ping to the Discord server.',
			throttling: {
				usages: 3,
				duration: 60
			},
			args: [
			{
				key: 'inp',
				prompt: '',
				type: 'integer',
				default: '1'
			}
			]
		});
	}

	async run(msg, {inp}) {
		if(inp==1){
			if(!msg.editable) {
				const pingMsg = await msg.reply('Pinging...');
				return pingMsg.edit(oneLine`
					${msg.channel.type !== 'dm' ? `${msg.author},` : ''}
					Pong! The message round-trip took ${pingMsg.createdTimestamp - msg.createdTimestamp}ms.
					${this.client.ping ? `The heartbeat ping is ${Math.round(this.client.ping)}ms...` : ''}
				`);
			}
			else {
				await msg.edit('Pinging...');
				return msg.edit(oneLine`
					Pong! The message round-trip took ${msg.editedTimestamp - msg.createdTimestamp}ms.
					${this.client.ping ? `The heartbeat ping is ${Math.round(this.client.ping)}ms.` : ''}
				`);
			}
		}
		else{
			var sum=0;
			for(var i=0; i<inp; i++)
			{
				const pingMsg = await msg.reply('Pinging...');
				pingMsg.edit(oneLine`
					${msg.channel.type !== 'dm' ? `${msg.author},` : ''}
					Pong! The message round-trip took ${pingMsg.createdTimestamp - msg.createdTimestamp - sum}ms.
					${this.client.ping ? `The heartbeat ping is ${Math.round(this.client.ping)}ms...` : ''}
				`);
				sum += pingMsg.createdTimestamp - msg.createdTimestamp;
			}
			msg.channel.send("Average ping is " + sum/inp);
		}
	}
};
